modules/geo-layers/src/tile-layer/tile-2d-header.js

```diff
@@ -17,12 +18,27 @@ export default class Tile2DHeader {
 
     this.onTileLoad = onTileLoad;
     this.onTileError = onTileError;
+
+    this.transformToWorld = null;
+    this._transformToWorldDataCache = null;
   }
 
   get data() {
     return this._isLoaded ? this.content : this._loader;
   }
 
+  get dataInWorldCoordinates() {
+    if (this.transformToWorld) {
+      if (this._transformToWorldDataCache === null) {
+        this._transformToWorldDataCache = this.content ? this.transformToWorld(this.content) : null;
```

Comment:
You need to await data if it's still loading. Currently this getter returns null with transform assigned, and a promise if it's not.